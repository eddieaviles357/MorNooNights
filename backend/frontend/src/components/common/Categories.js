import React from "react";
import "./Categories.css";

function Categories( { clickHandler } ) {
  function handleClick(evt) {
    // get category from dataset and pass data to clickHandler
    const {category} = evt.target.dataset;
    clickHandler(category, 1, true);
  }
  return (
    <>
    {/* <ul onClick={handleClick} className="Categories d-flex flex-column justify-content-evenly align-items-center flex-wrap p-2">
      <li data-category={"business"}>business</li>
      <li data-category={"entertainment"}>entertainment</li>
      <li data-category={"food"}>food</li>
      <li data-category={"politics"}>politics</li>
      <li data-category={"science"}>science</li>
      <li data-category={"sports"}>sports</li>
      <li data-category={"tech"}>tech</li>
      <li data-category={"travel"}>travel</li>
    </ul> */}
      <section 
        onClick={handleClick} 
        className="Categories d-flex flex-column justify-content-evenly align-items-center flex-wrap p-2">
        <div>
          <span data-category={"business"}>business</span>
          <span data-category={"entertainment"}>entertainment</span>
          <span data-category={"food"}>food</span>
          <span data-category={"politics"}>politics</span>
          <span data-category={"science"}>science</span>
          <span data-category={"sports"}>sports</span>
          <span data-category={"tech"}>tech</span>
          <span data-category={"travel"}>travel</span>
        </div>
      </section>
    </>
  )
}


export default Categories;