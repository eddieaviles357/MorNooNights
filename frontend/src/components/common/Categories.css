.Categories {
  background-color: var(--secondary);
  color: var(--stroke)
}

.Categories span {
  margin: 5px;
  cursor: pointer;
}

.Categories span:hover {
  color: var(--tertiary);
}