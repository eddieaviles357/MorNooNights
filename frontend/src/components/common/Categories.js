import React from "react";
import "./Categories.css";

function Categories( { clickHandler } ) {
  function handleClick(evt) {
    // get category from dataset and pass data to clickHandler
    const {category} = evt.target.dataset;
    clickHandler(category, 1, true);
  }
  return (
    <>
      <section 
        onClick={handleClick} 
        className="Categories d-flex flex-column justify-content-evenly align-items-center flex-wrap p-2">
        <div>
          <span data-category={"business"}>business</span>
          <span data-category={"entertainment"}>entertainment</span>
          <span data-category={"food"}>food</span>
          <span data-category={"politics"}>politics</span>
          <span data-category={"science"}>science</span>
          <span data-category={"sports"}>sports</span>
          <span data-category={"tech"}>tech</span>
          <span data-category={"travel"}>travel</span>
        </div>
      </section>
    </>
  )
}


export default Categories;