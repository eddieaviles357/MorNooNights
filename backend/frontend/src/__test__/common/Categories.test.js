import { render, cleanup } from '@testing-library/react';
import Categories from '../../components/common/Categories';
import { MemoryRouter } from "react-router";

afterAll(() => {
  cleanup();
});

describe("Categories component", () => {
  it('renders without crashing', () => {
    render(<Categories />);
  });

  it('matches snapshot', () => {
    const categories = render(<Categories />);
    expect(categories.asFragment()).toMatchSnapshot();
  });

  it('should be 8 categories', () => {
    const {getAllByText} = render(<Categories />);
    const entertainment = getAllByText("entertainment");
    const business = getAllByText("business");
    const politics = getAllByText("politics");
    const science = getAllByText("science");
    const sports = getAllByText("sports");
    const travel = getAllByText("travel");
    const food = getAllByText("food");
    const tech = getAllByText("tech");
    expect(entertainment.length).toBe(1);
    expect(business.length).toBe(1);
    expect(politics.length).toBe(1);
    expect(science.length).toBe(1);
    expect(sports.length).toBe(1);
    expect(travel.length).toBe(1);
    expect(food.length).toBe(1);
    expect(tech.length).toBe(1);
  });
});
