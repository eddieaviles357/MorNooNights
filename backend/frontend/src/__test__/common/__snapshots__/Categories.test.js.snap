// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Categories d-flex flex-column justify-content-evenly align-items-center flex-wrap p-2"
  >
    <div>
      <span
        data-category="business"
      >
        business
      </span>
      <span
        data-category="entertainment"
      >
        entertainment
      </span>
      <span
        data-category="food"
      >
        food
      </span>
      <span
        data-category="politics"
      >
        politics
      </span>
      <span
        data-category="science"
      >
        science
      </span>
      <span
        data-category="sports"
      >
        sports
      </span>
      <span
        data-category="tech"
      >
        tech
      </span>
      <span
        data-category="travel"
      >
        travel
      </span>
    </div>
  </section>
</DocumentFragment>
`;
